### PR TITLE
add Orders and get Orders through Post and Get methods.

### PR DESCRIPTION
**Pull request by Lazyness**
Create REST endpoint that will create and retrieve Orders to buy items throught POST and GET methods.
I hope i understood the task correctly =)
Use draft pull request

### CODE REVIEW

File: src/main/java/org/example/mentoringshop/controllers/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package org.example.mentoringshop.controllers;
+
+
+import org.example.mentoringshop.models.Order;
+import org.example.mentoringshop.services.OrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.awt.*;
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/controllers/OrderController.java:10]: I think it is useless import. You use nothing from java.awt

File: src/main/java/org/example/mentoringshop/controllers/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+@RestController
+@RequestMapping("orders")
+public class OrderController {
+    @Autowired
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/controllers/OrderController.java:15]: You don't have to use @Autowired here.
There are [Constructor-based dependency injection](https://docs.spring.io/spring-framework/docs/3.2.x/spring-framework-reference/html/beans.html#beans-constructor-injection) and [Setter-based dependency injection](https://docs.spring.io/spring-framework/docs/3.2.x/spring-framework-reference/html/beans.html#beans-setter-injection)

File: src/main/java/org/example/mentoringshop/controllers/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private OrderService orderService;
+
+    public OrderController(OrderService orderService) {
+        this.orderService = orderService;
+    }
+
+    @GetMapping
+    public Order getOrder(@RequestParam Integer id) {
+//        System.out.println("add id: " + id);
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/controllers/OrderController.java:24]: Do not save commented lines of code.

File: src/main/java/org/example/mentoringshop/controllers/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return orderService.getOrder(id);
+    }
+
+    @PostMapping (consumes = {
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/controllers/OrderController.java:28]: **consumes** is redundant now as we don't care what format a client uses - it all transforms to **Order**

File: src/main/java/org/example/mentoringshop/controllers/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            MediaType.APPLICATION_JSON_VALUE
+    },
+    produces = {
+            MediaType.APPLICATION_JSON_VALUE
+    })
+    public Order store(@RequestBody Order order) {
+//        System.out.println("add order: " + order);
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/controllers/OrderController.java:35]: Do not commit commented lines of code

File: src/main/java/org/example/mentoringshop/repositories/OrderRepository.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package org.example.mentoringshop.repositories;
+
+import org.example.mentoringshop.models.Order;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+
+@Component
+public class OrderRepository {
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/repositories/OrderRepository.java:9]: **OrderRepository should be an interface but not class**
We stick to DAO pattern that says " isolate the application/business layer from the persistence layer (usually a relational database but could be any other persistence mechanism) using an abstract API." [What is DAO Pattern in Java](https://www.baeldung.com/java-dao-pattern)
In our case OrderRepository is an abstract API and one of the implementations will be based on HashMap

File: src/main/java/org/example/mentoringshop/repositories/UserRepository.java [hunk 7]
```diff
@@ -22,6 +22,7 @@ public void save(User user) {
     }
 
     public User get(int index) {
+        System.out.println("user map.get(index): "+map.get(index));
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/repositories/UserRepository.java:25]: Do not commit System.out.println

File: src/main/java/org/example/mentoringshop/services/OrderService.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+@Service
+public class OrderService {
+    private OrderRepository orderRepository;
+
+    public OrderService(OrderRepository orderRepository) {
+        this.orderRepository = orderRepository;
+    }
+
+    public void addOrder(Order order) {
+//        System.out.println("save order: " + order);
```
Comment (well-another-attempt) [inline src/main/java/org/example/mentoringshop/services/OrderService.java:16]: Do not commit commented lines of code

**Review by well-another-attempt — COMMENTED** [review]
Ready

**Comment by Lazyness** [comment]
Ready. I resolve problem, please check

**Pull request merged**